static/js/pm_list.js

```diff
@@ -76,35 +133,143 @@ export function update_dom_with_unread_counts(counts) {
+
+    if (active_conversation) {
+        // If we find an active conversation thread satisfying the requirement, then we update our
+        // DOM with it's node.
+        const node = [pm_list_dom.keyed_pm_li(active_conversation)];
+        const new_dom = pm_list_dom.pm_ul(node);
+        const $container = ui.get_content_element($("#sidebar-private-messages-container"));
+        function replace_content(html) {
+            $container.html(html);
+        }
 
-    if (op_is.length >= 1 && op_is.includes("private")) {
-        return true;
+        function find() {
+            return $container.find("ul");
+        }
+        vdom.update(replace_content, find, new_dom, prior_dom);
+        prior_dom = new_dom;
+    } else {
+        // else we make the list empty and mark the section as collapsed.
+        $(".expanded_private_messages").empty();
+        close();
     }
+}
+
+export function show_all_private_messages_view() {
+    $("#private_messages_list").addClass("active_private_messages_list");
+    $("#more-private-conversations-sidebar-title").addClass(
+        "active-more-private-conversations-sidebar-title",
+    );
+}
+
+export function hide_all_private_messages_view() {
+    $("#private_messages_list").removeClass("active_private_messages_list");
+    $("#more-private-conversations-sidebar-title").removeClass(
+        "active-more-private-conversations-sidebar-title",
+    );
+}
 
-    const op_pm = filter.operands("pm-with");
+export function handle_narrow_activated(filter) {
+    const active_filter = filter;
 
-    if (op_pm.length !== 1) {
-        return false;
+    // we check for narrow to be of `all private messages view`, if not found,
+    // we remove the active class from `private_messages_list`.
+    if (!_.isEqual(active_filter.sorted_term_types(), ["is-private"])) {
+        hide_all_private_messages_view();
     }
 
-    const emails_strings = op_pm[0];
-    const emails = emails_strings.split(",");
+    if (active_filter.operands("pm-with").length !== 0) {
+        // if we find the narrow to be of a PM thread, we check for whether it belonging to
+        // collapsed/expanded section view and update the private messages section accordingly.
+        if (private_messages_collapsed) {
+            update_collapsed_private_messages_section();
+        }
+    } else {
+        // if the narrow is not related to private messages, we then deselect the PM section.
+        if (!_.isEqual(active_filter.sorted_term_types(), ["is-private"])) {
+            deselect_private_messages_section();
+        }
+    }
+}
 
-    const has_valid_emails = people.is_valid_bulk_emails_for_compose(emails);
+export function handle_narrow_deactivated() {
+    // while deactivaing the narrow, if private messages section is collapsed, we make the list empty.
+    if (private_messages_collapsed) {
+        $(".expanded_private_messages").empty();
```

Comment:
you don't need to call `$(".expanded_private_messages").empty();`, that happens in `close()`